libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.m

```diff
@@ -215,23 +215,21 @@ + (void)validateTimer
     }
 }
 
-+ (void)setInactivityConfiguration:(NSUInteger)newPasscodeLength lockoutTime:(NSUInteger)newLockoutTime biometricAllowed:(BOOL)newBiometricAllowed
-{
-    // No Passcode
-    if (newLockoutTime == 0) {
-        if (securityLockoutTime == 0) {
-            [SFSecurityLockout unlockSuccessPostProcessing:SFSecurityLockoutActionNone];
-        } else {
-            // '0' is a special case.  We can't turn off passcodes unless all other users' passcode policies
-            // are also off.
-            if (![SFSecurityLockout nonCurrentUsersHavePasscodePolicy]) {
-                [SFSecurityLockout clearAllPasscodeState];
-                [SFSecurityLockout unlock:YES action:SFSecurityLockoutActionPasscodeRemoved];
-                return;
-            }
-        }
++ (BOOL)doesNotNeedPasscodeFlow:(NSUInteger)newLockoutTime {
+    return (newLockoutTime == 0)  && ![self usersHavePasscodePolicy];
+}
+
++ (BOOL)hasNewPasscodePolicy:(NSUInteger)newLockoutTime passcodeLength:(NSUInteger)newPasscodeLength {
```

Comment:
Maybe we should call this something like `needsPasscodePolicyChange` instead. Anything entering this method already means there's a new login with a new passcode policy. We're only evaluating if the master passcode policy needs to change to reflect this new policy.